src/qibocal/auto/transpile.py

```diff
@@ -97,19 +98,65 @@ def execute_transpiled_circuit(
         backend,
         transpiler,
     )[0]
-
     return transpiled_circ, backend.execute_circuit(
         transpiled_circ, initial_state=initial_state, nshots=nshots
     )
 
 
+def get_natives(platform):
+    """
+    Return the list of native gates defined in the `platform`.
+    This function assumes the native gates to be the same for each
+    qubit and pair.
+    """
+    pairs = list(platform.pairs.values())[0]
+    qubit = list(platform.qubits.values())[0]
+    two_qubit_natives = list(pairs.native_gates.raw)
+    single_qubit_natives = list(qubit.native_gates.raw)
+    # Solve Qibo-Qibolab mismatch
+    single_qubit_natives.append("RZ")
+    single_qubit_natives.append("Z")
+    single_qubit_natives.remove("RX12")
+    replacements = {
+        "RX": "GPI2",
+        "MZ": "M",
+    }
+    new_single_natives = [replacements.get(i, i) for i in single_qubit_natives]
+    natives = new_single_natives + two_qubit_natives
+    return natives
+
+
+def set_compiler(backend, natives):
+    """
+    Set the compiler to execute the native gates defined by the platform.
+    """
+    compiler = backend.compiler
+    for native in natives:
+        gate = getattr(gates, native)
+        if gate not in compiler.rules:
+
+            def rule(qubits_ids, platform, parameters=None):
+                if len(qubits_ids[1]) == 1:
+                    native_gate = platform.qubits[tuple(qubits_ids[1])].native_gates
+                else:
+                    native_gate = platform.pairs[tuple(qubits_ids[1])].native_gates
+                pulses = getattr(native_gate, native).pulses
+                return PulseSequence(pulses), {}
+
+            backend.compiler[gate] = rule
+
+
 def dummy_transpiler(backend: Backend) -> Passes:
     """
     If the backend is `qibolab`, a transpiler with just an unroller is returned,
     otherwise None.
     """
-    unroller = Unroller(NativeGates.default())
-    return Passes(connectivity=backend.platform.topology, passes=[unroller])
+    platform = backend.platform
+    native_gates = get_natives(platform)
+    set_compiler(backend, native_gates)
+    native_gates = list(map(lambda x: getattr(gates, x), native_gates))
```

Comment:
Comprehension is doing the same, and it is much clearer to read:
```suggestion
    native_gates = [getattr(gates, x) for x in native_gates]
```

*(those languages heavily focusing on `map` and `lambda`, i.e. functional ones, are also those who conceived the comprehension itself)*